plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts

```diff
@@ -38,13 +38,21 @@ import { isBinaryFile } from 'isbinaryfile';
  */
 nunjucks.installJinjaCompat();
 
+type CookieCompatInput = {
+  copyWithoutRender?: string[];
+  cookiecutterCompat?: boolean;
+};
+
+type ExtensionInput = {
+  extension?: string | boolean;
```

Comment:
I don't mind naming much, but given that cookiecutter extensions are python modules, I don't think that the overlap in naming here would matter much.

Reply:
Hmm - the issue really for me is that the `fetch:cookiecutter` action (which is notionally very similar to this one) accepts an `extensions` option. As such if we go with `extension` here, we end up with almost identical option names in the two parallel actions that control totally different things. 

Re-reading my comment I see that I could have made this point a bit more clear - the link points to the option configuration in the `fetch:cookiecutter` action, but I definitely could have been more explicit!

Reply:
I think that a lot of ambiguity comes from having one action that tries to be cookiecutter compat and not. I chose a distinct action initially because I expected all these conflicts in interpretation.

To come back to the issue at hand, the docs for cookiecutter at https://cookiecutter.readthedocs.io/en/latest/advanced/template_extensions.html show

```
{
    "project_slug": "Foobar",
    "year": "{% now 'utc', '%Y' %}",
    "_extensions": ["jinja2_time.TimeExtension"]
}```

Looking at the value space here, I think that folks have a learning path ahead of them if they're getting this value confused with `.jinja2`.

I'm happy to name this whatever though. To me, this boils down to naming options of the `fetch:template` action with regards to cookiecutter for the upcoming years or not.

Reply:
Cool - I’m just concerned with clarity for consumers of these actions, particularly when they’re encountering them for the first time. How about we go with `templatedFileExtension`?

Reply:
I agree with the risk of confusion, with `templateFileExtension` being my pick. Just cause it's not the file extension that's being templated :grin: